Userland/Services/RequestServer/ConnectionFromClient.cpp

```diff
@@ -73,6 +74,18 @@ size_t ConnectionFromClient::on_header_received(void* buffer, size_t size, size_
     auto* request = static_cast<ActiveRequest*>(user_data);
     size_t total_size = size * nmemb;
     auto header_line = StringView { static_cast<char const*>(buffer), total_size };
+
+    // NOTE: We need to extract the HTTP reason phrase since it can be a custom value.
+    //       Fetching infrastructure needs this value for setting the status message.
+    if (header_line.starts_with("HTTP/"sv)) {
+        if (auto const space_positions = header_line.find_all(" "sv); space_positions.size() > 1) {
+            auto const second_space_offset = space_positions.at(1);
+            auto const reason_phrase_string_view = header_line.substring_view(second_space_offset + 1).trim_whitespace();
+            if (!reason_phrase_string_view.is_empty())
+                request->reason_phrase = MUST(String::from_utf8(reason_phrase_string_view));
```

Comment:
Is it guaranteed this is valid UTF-8?

Reply:
From quick research, it appears ISO-8859-1 is the encoding used for HTTP headers.

Reply:
Be careful with that, conflating ISO-8859-1 with UTF-8 is how we got into the mess in #1893

Reply:
And AFAIU, this is user-provided data, so we cannot trust it either. In other words, errors during decoding should be gracefully handled and not `MUST`ed.

Reply:
Isomorphic decoding is infallible because every input byte is mapped to a valid unicode codepoint

Reply:
All good then!